src/HIE/Bios/Cradle.hs

```diff
@@ -699,27 +699,39 @@ cabalAction
   -> FilePath
   -> CradleLoadResultT IO ComponentOptions
 cabalAction workDir mc l fp = do
-  cabalProc <- cabalProcess workDir "v2-repl" [fromMaybe (fixTargetPath fp) mc] `modCradleError` \err -> do
+  let
+    cabalCommand = "v2-repl"
+    cabalArgs = [fromMaybe (fixTargetPath fp) mc]
+
+  cabalProc <- cabalProcess workDir cabalCommand cabalArgs `modCradleError` \err -> do
       deps <- cabalCradleDependencies workDir workDir
       pure $ err { cradleErrorDependencies = cradleErrorDependencies err ++ deps }
 
   (ex, output, stde, [(_, maybeArgs)]) <- liftIO $ readProcessWithOutputs [hie_bios_output] l workDir cabalProc
-
   let args = fromMaybe [] maybeArgs
+
+  let errorDetails =
+        ["Failed command: " <> prettyCmdSpec (cmdspec cabalProc)
+        , unlines output
+        , unlines stde
+        , unlines $ args
+        , "Process Environment:"]
+        <> prettyProcessEnv cabalProc
+
+  when (ex /= ExitSuccess) $ do
+    deps <- liftIO $ cabalCradleDependencies workDir workDir
+    let cmd = show (["cabal", cabalCommand] <> cabalArgs)
+    let errorMsg = "Failed to run " <> cmd <> " in directory \"" <> workDir <> "\". See below for full command and error."
```

Comment:
I am not sure that "See below for full command and error." is going to be much helpful to users, since where is "below"?

Not sure this sentence is necessary.

Reply:
Maybe "See stderr for full command and error"?. I think it's helpful to point out that there's more information available

Reply:
I agree, but if this is shown in the LSP dialog pop-up, "below" has no meaning. However, I was actually unable to provoke that pop-up window, is this only shown as a diagnostic?

Reply:
I think you would need to build HLS with a `hie-bios` that includes https://github.com/haskell/hie-bios/pull/347 to get errors automatically shown to the user. I do agree that "below" is ambiguous here. Maybe "consult the logs"?

Reply:
@fendor Yes, it's only a diagnostic (on othe first line of the module). @michaelpj :+1: I changed the error msg like you suggested